game/objects/Portal.gd

```diff
@@ -0,0 +1,6 @@
+extends Area2D
+
+export (String, FILE, "*.tscn") var next_scene: String
+
+func _on_body_entered(body: PhysicsBody2D) -> void:
+	get_tree().change_scene(next_scene)
```

Comment:
In the metroidvania 2D project we started using `_get_configuration_warning` to warn the dev. to set a value on `export` vars when needed. Let's do that here too, it's a good habit to get into. I assume godot would complain if `next_scene` would be an empty string

Reply:
This feature is very buggy atm, and I think it doesn't provide that much value for small, beginner-friendly, **first game** project.

Reply:
Sure it provides value, it's exactly what people need in order to keep track of stuff that needs to be initialized. From the dev. point of view this is really useful, we're teaching here good habits with the idea of a growing team/game. It's good to introduce these habits from the very start. It's a very simple concept to get by the beginners - a warning system and it does a good job.

I haven't seen any bugs about the subject other than https://github.com/godotengine/godot/issues/26201 which is kind of a bug and not so I don't know what you mean by "is very buggy atm". Could you give me some examples of how this is so buggy we can't use it?

Reply:
I think I haven't had any bug with it either. Warnings appear as they should in the metroidvania

Reply:
> @NathanLovato in [GDquest/godot-platformer-2d#94](https://github.com/GDquest/godot-platformer-2d/pull/94) @guilhermehto mentions a bug, I also had some issues in my projects as well.
> 
I think what Gui is talking about is not a bug with `_get_configuration_warning`, but rather with `tool`: the refresh issue on save. Or rather not a bug, but more like a missing feature, if you close the scene & reopen it, it will refresh properly.

> Nonetheless, this feature is not a standalone addition and brings more than just few lines of code to the project:
> 
> * Introduce the `tool` mode
> * Explain `is_editor_hint()` and warn where this should be used when a script is using `tool` mode to make sure the code doesn't run in the editor
There will be no `is_editor_hint()` in the code and we won't need to explain it for this simple warning concept.

> 
> Plus, as I mentioned in the discord, this feature is mostly for working in teams, this project aims for beginners, making their first game, getting known to Godot, and this project will probably be followed by a tutorial series, which will explain what should be set in this property, that's what I mean with "this doesn't provide as much value as in the metroidvania", which is a complex project with a whole team working and no one with time to explain how their implementations should be used.
> 
> As I mentioned, I'd rather rely on the viewers' knowledge baggage, presented in #1, and show what they already can achieve with it, than introduce new concepts.
> 
> We should keep this simple, most of the audience won't need this feature by the time they get their hands on this project, is their first game, probably solo.

I think you're kind of getting the wrong idea about these tutorials. The key point is "good programming" not "programming with what you know". We're introducing concepts that help the developer reach a new level faster by explaining the benefit of certain systems and ways to think about coding.

Imperative programming offers an infinite amount of ways to structure code in order to solve a particular problem. For a beginner this isn't necessarily a good thing, that's why people end up having style guides in teams. We're teaching good habits from the start so that they know what's important down the line, we're not just teaching how to move some parts on the screen. That part is covered by every other sub-average tutorial out there. We want to do better and we'll use all the (easy to understand) systems that Godot offers at our disposal to do that. That's the point of using `_get_configuration_warning` here.

Reply:
> Imperative programming offers an infinite amount of ways to structure code in order to solve a particular problem. For a beginner this isn't necessarily a good thing, that's why people end up having style guides in teams. We're teaching good habits from the start so that they know what's important down the line, we're not just teaching how to move some parts on the screen. That part is covered by every other sub-average tutorial out there. We want to do better and we'll use all the (easy to understand) systems that Godot offers at our disposal to do that. That's the point of using _get_configuration_warning here.

Exactly that. The tutor is then here to break all of this down, make it as clear and as accessible as possible to the viewer.